### PR TITLE
[GNA] Multithread speech sample with iname

### PR DESCRIPTION
**Pull request by admitriev-gna**

### CODE REVIEW

File: inference-engine/samples/speech_sample/main.cpp [hunk 1]
```diff
@@ -1024,24 +1024,39 @@ int main(int argc, char *argv[]) {
...
-                                // locked memory holder should be alive all time while access to its buffer happens
-                                auto minputHolder = minput->wmap();
+                                ptrInputBlobs.push_back(blob);
+                            }
+                        }
 
-                                std::memcpy(minputHolder.as<void *>(),
-                                            inputFrame[i],
-                                            minput->byteSize());
+                        size_t num_files = FLAGS_iname.empty() ? numInputArkFiles : ptrInputBlobs.size();
```
Comment (kbruniec) [inline inference-engine/samples/speech_sample/main.cpp:1044]: Is there a chance that numInputArkFiles != ptrInputBlobs.size() ?
IMO there is not.

File: inference-engine/samples/speech_sample/main.cpp [hunk 2]
```diff
@@ -1024,24 +1024,39 @@ int main(int argc, char *argv[]) {
...
+                        for (size_t i = 0; i < num_files; ++i) {
+                            MemoryBlob::Ptr minput = as<MemoryBlob>(ptrInputBlobs[i]);
+                            if (!minput) {
+                                slog::err << "We expect ptrInputBlobs[" << i
```
Comment (kbruniec) [inline inference-engine/samples/speech_sample/main.cpp:1048]: why not throw logic_error, like in other places?

**Review by kbruniec — COMMENTED** [review]
No bug found
but...
@dorloff Definitely we need to refactor speech_sample,
because reviewing this change was much more complicated than it should be

**Comment by dorloff** [comment]
@kbruniec, it's well known that the speech sample requires refactoring, we are planning to do it.

**Review by kbruniec — APPROVED** [review]
LGTM

**Pull request merged**